api.c

```diff
@@ -0,0 +1,1055 @@
+        {"time-pos", required_argument, NULL, 540},
+        {"date-pos", required_argument, NULL, 541},
+        {"verif-pos", required_argument, NULL, 542},
+        {"wrong-pos", required_argument, NULL, 543},
+        {"layout-pos", required_argument, NULL, 544},
+        {"status-pos", required_argument, NULL, 545},
+        {"modif-pos", required_argument, NULL, 546},
+        {"ind-pos", required_argument, NULL, 547},
+        {"greeter-pos", required_argument, NULL, 548},
+
+        // text outline width
+        {"timeoutline-width", required_argument, NULL, 560},
+        {"dateoutline-width", required_argument, NULL, 561},
+        {"verifoutline-width", required_argument, NULL, 562},
+        {"wrongoutline-width", required_argument, NULL, 563},
+        {"modifieroutline-width", required_argument, NULL, 564},
+        {"layoutoutline-width", required_argument, NULL, 565},
+        {"greeteroutline-width", required_argument, NULL, 566},
+
+        // bar indicator stuff
+        {"bar-indicator", no_argument, NULL, 700},
+        {"bar-direction", required_argument, NULL, 701},
+        {"bar-orientation", required_argument, NULL, 703},
+        {"bar-step", required_argument, NULL, 704},
+        {"bar-max-height", required_argument, NULL, 705},
+        {"bar-base-width", required_argument, NULL, 706},
+        {"bar-color", required_argument, NULL, 707},
+        {"bar-periodic-step", required_argument, NULL, 708},
+        {"bar-pos", required_argument, NULL, 709},
+        {"bar-count", required_argument, NULL, 710},
+        {"bar-total-width", required_argument, NULL, 711},
+
+        {NULL, no_argument, NULL, 0}};
+
+    int o = 0;
+    int longoptind = 0;
+    char *optstring = "c:ui:tCFLMrs:kB:m";
+    char *arg = NULL;
+    int opt = 0;
+
+    char padded[9] = "ffffffff"; \
+
+#define parse_color(acolor)\
+    arg = optarg;\
+    if (arg[0] == '#') arg++;\
+    if (strlen(arg) == 6) {\
+      /* If 6 digits given, assume RGB and pad 0xff for alpha */\
+      strncpy( padded, arg, 6 );\
+      arg = padded;\
+    }\
+    if (strlen(arg) != 8 || sscanf(arg, "%08[0-9a-fA-F]", acolor) != 1) {\
+        fprintf(stderr, #acolor " is invalid, color must be given in 3 or 4-bye format: rrggbb[aa]\n");\
+        return;\
+    }
+
+#define parse_outline_width(awidth)\
+    arg = optarg;\
+    if (sscanf(arg, "%lf", &awidth) != 1) {\
+        fprintf(stderr, #awidth " must be a number.\n");\
+        return;\
+    }\
+    if (awidth < 0) {\
+        fprintf(stderr, #awidth " must be a positive double; ignoring...\n");\
+        awidth = 0;\
+    }
+
+#define apierr(fmt, ...) {\
+        fprintf(stderr, "[i3lock-error] " fmt, ##__VA_ARGS__);\
+        return;\
+    }
+
+    char *image_path = NULL;
+    char *image_raw_format = NULL;
+
+    optind = 0;
+
+    while ((o = getopt_long(argc, argv, optstring, longopts, &longoptind)) != -1) {
+        DEBUG("OPTLONG: %d: - %s\n", o, optarg);
+        switch (o) {
```

Comment:
Are these duplicated code from  i3lock.c?

Reply:
Yes, it's almost all duplicated, as "no big changes to main".